src/nublado/startup/services/preparer.py

```diff
@@ -0,0 +1,1193 @@
+            ),
+            "CULL_KERNEL_CONNECTED": "MappingKernelManager.cull_connected",
+            "CULL_KERNEL_INTERVAL": "MappingKernelManager.cull_interval",
+            "CULL_TERMINAL_INACTIVE_TIMEOUT": (
+                "TerminalManager.cull_inactive_timeout"
+            ),
+            "CULL_TERMINAL_INTERVAL": "TerminalManager.cull_interval",
+        }
+        result: list[str] = []
+        for envvar, setting in timeout_map.items():
+            val = self._env.get(envvar, "")
+            if val:
+                result.append(f"--{setting}={val}")
+        return result
+
+    def _write_lab_startup_files(self) -> None:
+        if self._broken:
+            self._logger.warning(
+                f"Abnormal startup: {self._env['ABNORMAL_STARTUP_MESSAGE']}"
+            )
+            self._make_abnormal_startup_environment()
+            #
+            # We will check to see if we got SCRATCH_DIR set before we broke,
+            # and if so, use that, which would be a user-specific path on a
+            # scratch filesystem.  If we didn't, we just use "/tmp" and hope
+            # for the best.  Any reasonably-configured RSP running under K8s
+            # will not have a shared "/tmp".
+            #
+            temphome = self._env.get("SCRATCH_DIR", "/tmp")
+            self._logger.warning(f"Launching with homedir='{temphome}'")
+            self._env["HOME"] = temphome
+            self._home = Path(temphome)
+        # Used by shell startup inside the Lab.
+        self._env["RUNNING_INSIDE_JUPYTERLAB"] = "TRUE"
+        # If any of these fails, lsst.rsp.startup ought to react to the
+        # lack of the appropriate files and start in degraded mode with
+        # an explanation.
+        try:
+            self._write_noninteractive_command()
+            self._write_lab_environment()
+            self._write_lab_args()
+        except Exception:
+            self._logger.exception("Writing Lab startup files failed")
+
+    def _write_noninteractive_command(self) -> None:
+        # Only write "noninteractive.json" if we are running in
+        # non-interactive mode.
+        if not bool(self._env.get("NONINTERACTIVE", "")):
+            return
+        config_path = (
+            get_runtime_mounts_dir()
+            / "noninteractive"
+            / "command"
+            / "command.json"
+        )
+        noninteractive_cmd_file = STARTUP_PATH / "noninteractive.json"
+        noninteractive_cmd_file.write_text(config_path.read_text())
+
+    def _write_lab_environment(self) -> None:
+        persistent_env = {
+            x: self._env[x]
+            for x in self._env
+            if x in RUNTIME_ENVIRONMENT_VARIABLES and self._env[x]
+        }
+        env_file = STARTUP_PATH / "env.json"
+        env_file.write_text(json.dumps(persistent_env))
+
+    def _write_lab_args(self) -> None:
+        log_level = "DEBUG" if self._debug else "INFO"
+        notebook_dir = f"{self._home!s}"
```

Comment:
I think since this is a Path object you could just do:
```suggestion
        notebook_dir = f"{self._home}"
```